src/discord-api.ts

```diff
@@ -141,3 +162,63 @@ const fetchData = async (url: string, token: string) =>
 			Accept: 'application/json',
 		},
 	})
+
+export const fetchAttachment = async (ctx: Context) => {
+	const request = ctx.req
+	const env = ctx.env
+
+	const origin = 'https://cdn.discordapp.com'
+	const attachment_url = new URL(origin + ctx.req.param('url'))
+
+	const file_name = attachment_url.pathname.split('/').pop() ?? ''
+
+	// Check memory cache first
+	const cached_url = cache.get(file_name)
+
+	if (cached_url && cached_url.expires.getTime() > Date.now()) {
+		return redirectResponse(
+			request,
+			cached_url.href,
+			cached_url.expires,
+			'memory'
+		)
+	}
+
+	const payload = {
+		method: 'POST',
+		headers: {
+			Authorization: `Bot ${env.DISCORD_TOKEN}`,
+			'Content-Type': 'application/json',
+		},
+		body: JSON.stringify({ attachment_urls: [attachment_url.href] }),
+	}
+
+	const response = await fetch(
+		`${DISCORD_URL}/attachments/refresh-urls`,
+		payload
+	)
+
+	// If failed return original Discord API response back
+	if (response.status != 200) {
+		return response
+	}
+
+	const json = await response.json<RefreshedResponse>()
+
+	if (Array.isArray(json?.refreshed_urls) && json.refreshed_urls[0].refreshed) {
+		const refreshed_url = new URL(json.refreshed_urls[0].refreshed)
+		const params = new URLSearchParams(refreshed_url.search)
+		// Convert from hex and add seconds
+		const expires = new Date(parseInt(params.get('ex') ?? '', 16) * 1000)
+
+		const cached_url: CachedURL = { href: refreshed_url.href, expires }
+
+		// Save to memory cache
+		cache.set(file_name, cached_url)
+
+		return redirectResponse(request, refreshed_url.href, expires, 'refreshed')
+	}
+
+	// Return Discord API json which does not have expected data
+	return Response.json(json, { status: 400 })
```

Comment:
Perhaps a 404 would make more sense here?
```suggestion
	return Response.json(json, { status: 404 })
```